src/components/Widget.tsx

```diff
@@ -114,10 +114,27 @@ export default function Widget(props: PropsWithChildren<WidgetProps>) {
     }
     return props.locale ?? DEFAULT_LOCALE
   }, [props.locale])
-  const jsonRpcEndpoint = useMemo(
-    () => props.jsonRpcEndpoint ?? JSON_RPC_FALLBACK_ENDPOINTS[SupportedChainId.MAINNET]?.[0],
-    [props.jsonRpcEndpoint]
-  )
+  const defaultChainId = useMemo(() => {
+    if (!props.defaultChainId) return 1
+    if (!ALL_SUPPORTED_CHAIN_IDS.includes(props.defaultChainId)) {
+      console.warn(`Unsupported chainId: ${props.defaultChainId}. Falling back to 1 (Ethereum Mainnet).`)
+      return 1
+    }
+    return props.defaultChainId
+  }, [props.defaultChainId])
+  const jsonRpcUrlMap: string | JsonRpcProvider | { [chainId: number]: string[] } = useMemo(() => {
```

Comment:
at this point can't we just make sure it's only of type `{ [chainId: number]: string[] }`?